compiler/ilgen/MethodBuilder.cpp

```diff
@@ -331,6 +331,8 @@ MethodBuilder::symbolDefined(const char *name)
 void
 MethodBuilder::defineSymbol(const char *name, TR::SymbolReference *symRef)
    {
+   TR_ASSERT_FATAL(_symbols.find(name) == _symbols.end(), "Symbol '%s' already defined", name);
```

Comment:
in future, commits like this one could be made in its own pull request. As it stands, this commit doesn't *really* fit in with the title of the MR.